File: app/src/main/java/com/sortedqueue/programmercreek/util/PrettifyHighlighter.java
package com.sortedqueue.programmercreek.util;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

import prettify.PrettifyParser;
import syntaxhighlight.ParseResult;
import syntaxhighlight.Parser;

public class PrettifyHighlighter {
    //private static final Map<String, String> COLORS = buildColorsMap_Dark_BG();
    private static final Map<String, String> COLORS = buildColorsMap_White_BG();

    private static final String FONT_PATTERN = "<font color=\"#%s\">%s</font>";

    private final Parser parser = new PrettifyParser();
    
    static PrettifyHighlighter mInstance;
    public static PrettifyHighlighter getInstance() {
    	if( mInstance == null ) {
    		mInstance = new PrettifyHighlighter();
    	}
    	return mInstance;
    }

    public String highlight(String fileExtension, String sourceCode) {
        StringBuilder highlighted = new StringBuilder();
        List<ParseResult> results = parser.parse(fileExtension, sourceCode);
        for(ParseResult result : results){
            String type = result.getStyleKeys().get(0);
            String content = sourceCode.substring(result.getOffset(), result.getOffset() + result.getLength());
            highlighted.append(String.format(FONT_PATTERN, getColor(type), content));
        }
        return highlighted.toString();
    }

    private String getColor(String type){
        return COLORS.containsKey(type) ? COLORS.get(type) : COLORS.get("pln");
    }

    @SuppressWarnings("unused")
	private static Map<String, String> buildColorsMap_Dark_BG() {
        Map<String, String> map = new HashMap<String, String>();
        map.put("typ", "87cefa"); //#517C96
        map.put("kwd", "00ff00"); //#006600
        map.put("lit", "ffff00"); //#6B6B00
        map.put("com", "999999"); //#5C5C5C
        map.put("str", "ff4500"); //#ff4500
        map.put("pun", "eeeeee"); //#8F8F8F
        map.put("pln", "ffffff"); //#000000
        return map;
    }
    
    private static Map<String, String> buildColorsMap_White_BG() {
        Map<String, String> map = new HashMap<String, String>();
        map.put("typ", "517C96"); //#517C96
        map.put("kwd", "006600"); //#006600
        map.put("lit", "006699"); //#6B6B00
        map.put("com", "6B6B00"); //#5C5C5C
        map.put("str", "ff4500"); //#ff4500
        map.put("pun", "ff4500"); //#8F8F8F
        map.put("pln", "000000"); //#000000
        return map;
    }
}

File: app/src/main/java/com/sortedqueue/programmercreek/util/PrettifyHighlighter.kt
package com.sortedqueue.programmercreek.util

import java.util.HashMap

import prettify.PrettifyParser
import syntaxhighlight.ParseResult
import syntaxhighlight.Parser

class PrettifyHighlighter {

    private val parser = PrettifyParser()

    fun highlight(fileExtension: String, sourceCode: String): String {
        val highlighted = StringBuilder()
        val results = parser.parse(fileExtension, sourceCode)
        for (result in results) {
            val type = result.styleKeys[0]
            val content = sourceCode.substring(result.offset, result.offset + result.length)
            highlighted.append(String.format(FONT_PATTERN, getColor(type), content))
        }
        return highlighted.toString()
    }

    private fun getColor(type: String): String {
        return if (COLORS.containsKey(type)) COLORS[type]!! else COLORS["pln"]!!
    }

    companion object {
        //private static final Map<String, String> COLORS = buildColorsMap_Dark_BG();
        private val COLORS = buildColorsMap_White_BG()

        private val FONT_PATTERN = "<font color=\"#%s\">%s</font>"

        internal var mInstance: PrettifyHighlighter? = null
        val instance: PrettifyHighlighter
            get() {
                if (mInstance == null) {
                    mInstance = PrettifyHighlighter()
                }
                return mInstance!!
            }

        private fun buildColorsMap_Dark_BG(): Map<String, String> {
            val map = HashMap<String, String>()
            map.put("typ", "87cefa") //#517C96
            map.put("kwd", "00ff00") //#006600
            map.put("lit", "ffff00") //#6B6B00
            map.put("com", "999999") //#5C5C5C
            map.put("str", "ff4500") //#ff4500
            map.put("pun", "eeeeee") //#8F8F8F
            map.put("pln", "ffffff") //#000000
            return map
        }

        private fun buildColorsMap_White_BG(): Map<String, String> {
            val map = HashMap<String, String>()
            map.put("typ", "517C96") //#517C96
            map.put("kwd", "006600") //#006600
            map.put("lit", "006699") //#6B6B00
            map.put("com", "6B6B00") //#5C5C5C
            map.put("str", "ff4500") //#ff4500
            map.put("pun", "ff4500") //#8F8F8F
            map.put("pln", "000000") //#000000
            return map
        }
    }
}